app/src/main/java/com/github/bjolidon/bootcamp/database/UserCardsRTDB.kt

```diff
@@ -0,0 +1,72 @@
+package com.github.bjolidon.bootcamp.database
+
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import com.github.bjolidon.bootcamp.R
+import com.github.bjolidon.bootcamp.model.MagicCard
+import com.github.bjolidon.bootcamp.model.MagicLayout
+import com.github.bjolidon.bootcamp.model.MagicSet
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+import java.util.concurrent.CompletableFuture
+
+/**
+ * This class is used to manage the user's collection of cards.
+ */
+
+class UserCardsRTDB : AppCompatActivity(){

```

Comment:
does it need to be an activity? there is no UI associated ?

Reply:
oh yeah most likely no, the point of UserCardsRTDB was to hold the functions responsible for fetching the actual data (so nothing to do with ui). This allows these functions to be useful later on if we decide we need to access the database somewhere

Reply:
then i would make it a class only, not an Activity